### PR TITLE
feat(MeshExternalService): implement validator

### PR DESCRIPTION
**Pull request by slonka**
checking if everything passes CI

### Checklist prior to review

- [ ] [Link to relevant issue][1] as well as docs and UI issues --
- [ ] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [ ] Tests (Unit test, E2E tests, manual test on universal and k8s) --
  - Don't forget `ci/` labels to run additional/fewer tests
- [ ] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [ ] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? ([this](https://github.com/kumahq/kuma/actions/workflows/auto-backport.yaml) GH action will add "backport" label based on these [file globs](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L6), if you want to prevent it from adding the "backport" label use [no-backport-autolabel](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L8) label) --

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: pkg/core/resources/apis/meshexternalservice/api/v1alpha1/meshexternalservice.go [hunk 1]
```diff
@@ -57,44 +68,59 @@ type Endpoint struct {
 	// +kubebuilder:validation:MinLength=1
 	Address string `json:"address"`
 	// Port of the endpoint
-	Port EndpointPort `json:"port"`
+	Port *EndpointPort `json:"port,omitempty"`
```
Comment (lukidzi) [inline pkg/core/resources/apis/meshexternalservice/api/v1alpha1/meshexternalservice.go:71]: Should it be an `EndpointPort` if it's in the match? Maybe just `Port` is enough

File: pkg/core/resources/apis/meshexternalservice/api/v1alpha1/meshexternalservice.go [hunk 2]
```diff
@@ -57,44 +68,59 @@ type Endpoint struct {
...
 	// +kubebuilder:default=false
-	AllowRenegotiation bool `json:"allowRenegotiation"`
+	AllowRenegotiation bool `json:"allowRenegotiation,omitempty"`
 	// Verification section for providing TLS verification details.
-	Verification Verification `json:"verification"`
+	Verification *Verification `json:"verification,omitempty"`
 }
 
 // +kubebuilder:validation:Enum=TLSAuto;TLS10;TLS11;TLS12;TLS13
 type TlsMinMaxVersion string
```
Comment (lukidzi) [inline pkg/core/resources/apis/meshexternalservice/api/v1alpha1/meshexternalservice.go:89]: Maybe just TlsVersion

File: pkg/core/resources/apis/meshexternalservice/api/v1alpha1/validator.go [hunk 3]
```diff
@@ -1,10 +1,144 @@
...
+
 	return verr.OrNil()
 }
+
+func validateTls(tls *Tls) validators.ValidationError {
+	var verr validators.ValidationError
+
+	if tls.Version != nil {
+		path := validators.RootedAt("version")
+		if tls.Version.Min != nil {
```
Comment (lukidzi) [inline pkg/core/resources/apis/meshexternalservice/api/v1alpha1/validator.go:49]: Should we validate cases:
- min is Auto and max is specific
- max is Auto and min is specific
- min is higher than max

File: pkg/core/resources/apis/meshexternalservice/api/v1alpha1/validator.go [hunk 4]
```diff
@@ -1,10 +1,144 @@
...
+			}
+		}
+		if tls.Version.Max != nil {
+			if !slices.Contains(allTlsVersions, string(*tls.Version.Max)) {
+				verr.AddErrorAt(path.Field("max"), validators.MakeFieldMustBeOneOfErr("max", allTlsVersions...))
+			}
+		}
+	}
+
+	if tls.Verification != nil {
```
[the remark this answers is not in the document]
Comment (slonka) [reply pkg/core/resources/apis/meshexternalservice/api/v1alpha1/validator.go:61]: why? if you have enabled = false then we just not apply any TLS settings, it's not invalid to have enabled=true and verification != nil, right?

[the remark this answers is not in the document]
Comment (slonka) [reply pkg/core/resources/apis/meshexternalservice/api/v1alpha1/validator.go:51]: idea was to not do validation if `enabled=false` -> but we decided offline not to do this

**Review by lukidzi — APPROVED** [review]

**Pull request merged**